core/textile/bucket_factory.go

```diff
@@ -119,12 +118,19 @@ func (tc *textileClient) getOrCreateBucketContext(ctx context.Context, bucketSlu
 	log.Debug("getOrCreateBucketContext: Thread ID not found in meta store. Generating a new one...")
 	dbID := thread.NewIDV1(thread.Raw, 32)
 
+	managedKey, err := tc.kc.GetManagedThreadKey()
+	if err != nil {
+		log.Error("error getting managed thread key", err)
+		return nil, nil, err
+	}
+
 	log.Debug("getOrCreateBucketContext: Creating Thread DB for bucket " + bucketSlug + " at db " + dbID.String())
-	if err := tc.threads.NewDB(ctx, dbID); err != nil {
+	if err := tc.threads.NewDB(ctx, dbID, db.WithNewManagedThreadKey(managedKey)); err != nil {
```

Comment:
What's this for?

Reply:
@dmerrill6 

My intuition was to manage keys for every thread that's not public. What do you think, @jsonsivar ?

Reply:
Before approving, need to fully understand what's this for since you are using the public key for this value, but given it's public it can be generated by anyone which I'm not sure if it brings issues like collisions. If you check `textile/utils` in the method `NewDeterministicThreadID` we derive the thread id using a KDF from the private key, which would solve this issue if it is.

Reply:
Yeah the thread key should be the private key or something derived from it but not a public key. @maurycy saw the fix commit for this but I think that should be using the private key instead of the public key.

Reply:
@jsonsivar 

We will be syncing about this later today, because it turned out that `NewDeterministicThreadID` is not using the private key as well. :-)

Reply:
I think it might be okay that the thread ID creation is using the pubkey since that is just the thread ID.  Even if someone tries to generate it using the pubkey and then tries accessing, it wont give them access because it was created with the private key context.  Here we are creating a thread key (not thread ID) hence the difference and need to definitely use the private key.  Just clarifying the difference, but we can still make the thread ID also use the private key as well for consistency but it is 2 different things we are doing here.

Reply:
@jsonsivar 

What do you think? https://github.com/FleekHQ/space-daemon/pull/191/commits/2bf0da6c2f418e5412ed4f632666eae22c6ade2e#diff-a509c084f1fb203e2d90b6bd7e71b528R377

Reply:
Nice looks good.